torch/csrc/jit/ir.cpp

```diff
@@ -1328,6 +1328,23 @@ void Node::removeFromList() {
   this->prev() = nullptr;
 }
 
+Block* Node::findCommonAncestorBlockWith(Node* n) {
+  std::unordered_set<const Block*> seen_blocks_rhs;
+
+  for (Block* rhs_block = n->owningBlock(); rhs_block;
```

Comment:
I believe a co-iteration over both chains simultaneously would be more efficient in most cases, because you'll end up not having to construct the full set.

Reply:
I believe this function does it correctly: https://github.com/pytorch/pytorch/blob/f7b2778cb1db148f1c901ea08abcbfa0e0a78f44/torch/csrc/jit/passes/canonicalize.cpp#L91